azavea/district-builder-dtl-pa
# Add locustfile with real-world request proportions

## Overview

Updates the locustfile to mimic the activity recorded during a 5-minute redistricting session.

### Checklist


### Notes

- In the interest of time, I ignored any URL patterns that had fewer than 10 requests within the 5-minute testing span. I'm attaching the raw data so that these can be added later if necessary.
[requests.csv.txt](https://github.com/azavea/district-builder-dtl-pa/files/1981366/requests.csv.txt)

## Testing Instructions

 * Bring up your VM and then `./scripts/server`
 * In another terminal, `vagrant ssh` and then `locust`
 * Browse to `http://localhost:8089` and start swarming.
 * Double-check that cookies are working in the way this expects and that the requests are actually testing the real operations and not short-circuiting because of auth issues; I *think* I got everything right and the API seems to be returning 200s, but I don't totally trust it not to return 200s but still be failing. You can comment out different TaskSets in [locustfile.py](https://github.com/azavea/district-builder-dtl-pa/compare/develop...azavea:feature/load-test-setup-%23157109250?expand=1#diff-3f575e12833c89c68a25627dec755f93R404) to make this easier.
